File: remonTest1/remonTest1/ViewController.m
//
//  ViewController.m
//  remonTest1
//
//  Created by dnasoft on 2015. 1. 22..
//  Copyright (c) 2015년 dnasoft. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (nonatomic, strong) NSMutableArray *listArray;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.navigationItem.title = @"Remon SDK Sample";
    
    NSMutableArray *listArray = [[NSMutableArray alloc] initWithObjects:@"Inline", @"Square", @"Rectangle", @"Floating", @"Interstitial", @"Agreement", nil];
    
    [self setListArray:listArray];
    listArray = nil;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark -
#pragma mark UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    return _listArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell"];
    
    cell.textLabel.text = [_listArray objectAtIndex:indexPath.row];
    
    return cell;
}

#pragma mark -
#pragma mark UITableViewDelegate

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    return 50.0f;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UIViewController *vc;
    
    switch (indexPath.row) {
        case 0:
            vc = [self getViewController:@"InlineViewController"];
            break;
        case 1:
            vc = [self getViewController:@"SquareViewController"];
            break;
        case 2:
            vc = [self getViewController:@"RectangleViewController"];
            break;
        case 3:
            vc = [self getViewController:@"FloatingViewController"];
            break;
        case 4:
            vc = [self getViewController:@"InterstitialViewController"];
            break;
        case 5:
            vc = [self getViewController:@"AgreementViewController"];
            break;
        default:
            break;
    }
    
    if(vc != nil) {
        [vc setModalTransitionStyle:UIModalTransitionStyleCoverVertical];
        [self.navigationController pushViewController:vc animated:YES];
    }
}

- (UIViewController *) getViewController:(NSString *)idenifier {
    
    UIViewController * vc = [self.storyboard instantiateViewControllerWithIdentifier:idenifier];
    
    return vc;
}

@end

File: remontest_swift/remontest_swift/ViewController.swift
//
//  ViewController.swift
//  remontest_swift
//
//  Created by Rabbit on 2015. 1. 21..
//  Copyright (c) 2015년 dnasoft. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    @IBOutlet weak var tableView: UITableView!
    var listArray = NSMutableArray()
    let tableData = ["Inline", "Square", "Rectangle", "Floating", "Interstitial", "Agreement"]

    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        for index in 1...5 {
            listArray.addObject(index)
        }
        
        self.navigationItem.title = "Remon SDK Sample"
        
        self.tableView.registerClass(UITableViewCell.classForCoder(), forCellReuseIdentifier:"Cell")

        
        println("id : \(NSBundle.mainBundle().infoDictionary)")

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return self.tableData.count;
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return 50.0;
    }
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1;
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath) as UITableViewCell
        cell.textLabel?.text = "\(self.tableData[indexPath.row])"
        return cell
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
    
        //openModal(indexPath.row);
        var vc:UIViewController
        
        switch indexPath.row {
        case 0:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("InlineViewController") as UIViewController
        case 1:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("SquareViewController") as UIViewController
        case 2:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("RectangleViewController") as UIViewController
        case 3:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("FloatingViewController") as UIViewController
        case 4:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("InterstitialViewController") as UIViewController
        case 5:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("AgreementViewController") as UIViewController
        default:
            vc = self.storyboard?.instantiateViewControllerWithIdentifier("InlineViewController") as UIViewController
        }
        
        //var vc:UIViewController = self.storyboard?.instantiateViewControllerWithIdentifier("InlineViewController") as UIViewController
        vc.modalTransitionStyle = UIModalTransitionStyle.CoverVertical
        //self.navigationController?.pushViewController(vc, animated: true)
        //var nc: UINavigationController = self.navigationController!
         self.navigationController?.pushViewController(vc, animated: true)
    }
    
    func openModal(inde: Int) {
        
        presentViewController(InlineViewController(), animated: false, completion: nil);
    }

}